### ISSUE TITLE
Gtk2 graphem problems on windows

### ISSUE THREAD

**Issue by ProvGANG** [Shoes3/shoes3#96]
![shoes-win](https://cloud.githubusercontent.com/assets/10463333/6895119/c67fe030-d6dc-11e4-9eb6-78aabbfce335.png)

![shoes-linux](https://cloud.githubusercontent.com/assets/10463333/6895147/17077bc6-d6dd-11e4-8b90-fd678467120d.png)

These are two pics that show how it works on win and linux. So the problem is on windows as it can't represent the graphems in a way that I'd want to.

**Comment by BackOrder** [comment]
It might simply be the fonts used by default is not UTF8 friendly.

**Comment by ProvGANG** [comment]
Tried several common fonts, it's not about it. Otherwise the same problem would've appeared on linux too.

**Comment by ccoupe** [comment]
I suspect the font and the new fontconfig code and/or cache are confused. What is the font(s) names? Truetype? or ?

**Comment by ccoupe** [comment]
The following script displays the proper characters in my en.us Linux and Windows using the default font for each - Arial for Windows I believe.
```ruby
Shoes.app do
  stack do
    para "Display some croation characters\n"
    title "\u0161 \u0111 \u010d \u0107 \u017e"
  end
end
```

**Comment by ccoupe** [comment]
Updated the test script above to add an edit_line. It works fine for me using the Linux keystrokes for unicode entry

**Comment by ProvGANG** [comment]
Guys is there an easy way to edit last 3.2.21 version of shoes to be able to put :title to alert widget? I'm sorry for being little out of topic but last version of Shoes was okay considering such encoding problems as new Win version is having, I just want to have that update, any solutions?

**Comment by passenger94** [comment]
if you're able to compile shoes, just copy/paste relevant methods near the end of gtk.c, should work and the most simple i think

you could also try, as an exercise in eccentricities, some awful ludicrous aerobics by copying libshoes from 3.2.22 to 3.2.21

**Comment by ccoupe** [comment]
@ProvGANG,  you could try moving the 3.2.22 libshoes.dll to 3.2.21. It probably won't work. It would be better for everyone if you could help us debug why 3.2.22 doesn't work on your system. Any future version of Shoes is going to have the same problem if we don't debug it.

 Does the script above display the characters? Can you enter them and are they echoed back correctly. We need to separate display (font) and keyboard/input issues.

**Comment by ccoupe** [comment]
Coincidence - someone just file https://github.com/shoes/shoes/issues/272 at the old shoes repo.

**Comment by ProvGANG** [comment]
The same thing persists when running the above code, here is the comparision when running older version and 3.2.22 version. Replacing libshoes.dll worked, but I agree that we should try to get this fixed. It could be the font issue, as I declare a font on 3.2.22 version it stays the same as on default you can see on the image. 

![shoes](https://cloud.githubusercontent.com/assets/10463333/6999056/d37db048-dbf8-11e4-90dc-86541b159e88.png)

**Comment by ProvGANG** [comment]
Also if it helps, when I run cshoes from cmd, I get this message: 
Fontconfig error: Cannot load default config file

**Comment by ccoupe** [comment]
Yes that error message does help! That is an important clue.  Lets see if we can rebuild the font cache or get more messages.  In  your HOME APPDATA\Local should be a fontconfig folder. Delete it and from the cmd line run cshoes.exe - it should take two minutes or so without errors. 

What font is set as your Windows default?

**Comment by ProvGANG** [comment]
Alright, I've deleted fontonfig, error message is still the same when I run
cshoes.exe

2015-04-06 2:02 GMT+02:00 Cecil Coupe <notifications@github.com>:

> Yes that error message does help! That is an important clue. Lets see if
> we can rebuild the font cache or get more messages. In your HOME
> APPDATA\Local should be a fontconfig folder. Delete it and from the cmd
> line run cshoes.exe - it should take two minutes or so without errors.
>
> What font is set as your Windows default?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Shoes3/shoes3/issues/96#issuecomment-89870777>.
>



-- 
*Cheers!*

**Comment by ccoupe** [comment]
What font/language settings are you using for Windows. I need to duplicate them to see if the bug happens here.

**Comment by ccoupe** [comment]
Are you installing Shoes in the default locations c:\Program Files (x86)\Shoes ?   I know its a lot of questions but I have to ask what I don't know.

**Comment by ccoupe** [comment]
Luka, do you have both 3.2.21 and 3.2.22 installed?  If so the env var may not be set properly

I need to know where your Shoes 3.2.22 is installed and what FONTCONFIG_FILE is set too.
```
C:\Users\Cecil>set FONTCONFIG_FILE
FONTCONFIG_FILE=C:\Program Files (x86)\Shoes\etc\fonts\fonts.conf
```

**Comment by ccoupe** [comment]
Please see https://github.com/Shoes3/shoes3/issues/106#issuecomment-89932767

**Comment by ProvGANG** [comment]
I had comma separated values in ENV variable, deleted it, changed it to FONTCONFIG_FILE=C:\Program Files (x86)\Shoes\etc\fonts\fonts.conf 

And what actually the problem is that I had old version(from November 2014) of Shoes installed in different directory which I knew was there but totally forgot about it and shoes was running from there.

I'm not sure why it hasn't caused any problem prior to this release but now everything's okay.

**Comment by ccoupe** [comment]
I'm going to close this issue since 1) the problem was fixed, 2) #106 is similar and open.
